### DISCUSSION TITLE
fluentd-ui start command gives error - bundler: failed to load command: rackup

### DISCUSSION THREAD

**Discussion by avi-git-lab** [fluent/fluentd#4402] [Q&A]
Hi Team,
I am getting below error after executing "fluentd-ui start" command.

$ fluentd-ui start
bundler: failed to load command: rackup (/home/avi/.rbenv/versions/3.3.0/bin/rackup)
/home/avi/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/activesupport-5.2.8.1/lib/active_support/core_ext/module/delegation.rb:156:in `delegate': Delegation needs a target. Supply an options hash with a :to key as the last argument (e.g. delegate :hello, to: :greeter). (ArgumentError)

      raise ArgumentError, "Delegation needs a target. Supply an options hash with a :to key as the last argument (e.g. delegate :hello, to: :greeter)."

I have below configuration in my setup 
=========================================
01) OS details
OS: Ubuntu 23.10
# ufw status
Status: inactive
# sestatus
SELinux status:                 disabled

02) Ruby installation- 
I installed ruby using below sequential commands-

sudo apt update
sudo apt install git curl autoconf bison build-essential libssl-dev libyaml-dev libreadline6-dev zlib1g-dev libncurses5-dev libffi-dev libgdbm6 libgdbm-dev libdb-dev
curl -fsSL https://github.com/rbenv/rbenv-installer/raw/HEAD/bin/rbenv-installer | bash
rbenv install 3.3.0
rbenv global 3.3.0
export PATH=$HOME/.rbenv/versions/3.3.0/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin

ruby -v
ruby 3.3.0 (2023-12-25 revision 5124f9ac75) [x86_64-linux]

cd $HOME/.rbenv/versions/3.3.0/bin
gem update builder
gem update --system

03) fluentd-ui Installation -
 gem install -V fluentd-ui
 fluentd-ui setup 

04) start fluentd-ui
 fluentd-ui start
 At this command I am getting above mentioned error.

05) Troubleshooting efforts -
    a> Tried by uninstalling and re-installing fluentd-ui
        gem uninstall fluentd-ui
        gem install fluentd-ui
    b> Tried installation using above steps 01 to 04 on ubuntu 20.04, 22.04, 23.10
       But the faced the same error.

I have never worked in Ruby development hence not able to fix the issue. Attaching full error for your reference. Please let me know if any further details required.
[fluentd-ui start error.txt](https://github.com/fluent/fluentd/files/14295288/fluentd-ui.start.error.txt)

Kindly help.

**Comment by Alisher-Nabiev** [comment]
hi @avi-git-lab did u resolve the issue i am getting something similar ?
`/var/lib/gems/3.0.0/gems/activesupport-5.2.8.1/lib/active_support/core_ext/module/delegation.rb:156:in `delegate': Delegation needs a target. Supply an options hash with a :to key as the last argument (e.g. delegate :hello, to: :greeter). (ArgumentError)`

**Comment by Thaleikis** [comment]
Having the same issue on Debian12:
`bundler: failed to load command: rackup (/usr/local/bin/rackup)
/var/lib/gems/3.1.0/gems/activesupport-5.2.8.1/lib/active_support/core_ext/module/delegation.rb:156:in `delegate': Delegation needs a target. Supply an options hash with a :to key as the last argument (e.g. delegate :hello, to: :greeter). (ArgumentError)
	from /var/lib/gems/3.1.0/gems/draper-3.1.0/lib/draper/delegation.rb:10:in `delegate'
	from /var/lib/gems/3.1.0/gems/activemodel-5.2.8.1/lib/active_model/naming.rb:222:in `extended'
	from /var/lib/gems/3.1.0/gems/activemodel-5.2.8.1/lib/active_model/serializers/json.rb:13:in `extend'
	from /var/lib/gems/3.1.0/gems/activemodel-5.2.8.1/lib/active_model/serializers/json.rb:13:in `block in <module:JSON>'
	from /var/lib/gems/3.1.0/gems/activesupport-5.2.8.1/lib/active_support/concern.rb:122:in `class_eval'
	from /var/lib/gems/3.1.0/gems/activesupport-5.2.8.1/lib/active_support/concern.rb:122:in `append_features'
	from /var/lib/gems/3.1.0/gems/draper-3.1.0/lib/draper/decorator.rb:10:in `include'
	from /var/lib/gems/3.1.0/gems/draper-3.1.0/lib/draper/decorator.rb:10:in `<class:Decorator>'
	from /var/lib/gems/3.1.0/gems/draper-3.1.0/lib/draper/decorator.rb:4:in `<module:Draper>'
	from /var/lib/gems/3.1.0/gems/draper-3.1.0/lib/draper/decorator.rb:3:in `<main>'
	from /var/lib/gems/3.1.0/gems/bootsnap-1.18.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30:in `require'
	from /var/lib/gems/3.1.0/gems/bootsnap-1.18.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30:in `require'
	from /var/lib/gems/3.1.0/gems/draper-3.1.0/lib/draper.rb:18:in `<main>'
	from /var/lib/gems/3.1.0/gems/bootsnap-1.18.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30:in `require'
	from /var/lib/gems/3.1.0/gems/bootsnap-1.18.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30:in `require'
	from /var/lib/gems/3.1.0/gems/fluentd-ui-1.2.1/config/application.rb:21:in `<top (required)>'
	from /var/lib/gems/3.1.0/gems/fluentd-ui-1.2.1/config/environment.rb:2:in `require_relative'
	from /var/lib/gems/3.1.0/gems/fluentd-ui-1.2.1/config/environment.rb:2:in `<top (required)>'
	from /var/lib/gems/3.1.0/gems/fluentd-ui-1.2.1/config.ru:3:in `require'
	from /var/lib/gems/3.1.0/gems/fluentd-ui-1.2.1/config.ru:3:in `block in <main>'
	from /var/lib/gems/3.1.0/gems/rack-2.2.9/lib/rack/builder.rb:116:in `eval'
	from /var/lib/gems/3.1.0/gems/rack-2.2.9/lib/rack/builder.rb:116:in `new_from_string'
	from /var/lib/gems/3.1.0/gems/rack-2.2.9/lib/rack/builder.rb:105:in `load_file'
	from /var/lib/gems/3.1.0/gems/rack-2.2.9/lib/rack/builder.rb:66:in `parse_file'
	from /var/lib/gems/3.1.0/gems/rack-2.2.9/lib/rack/server.rb:349:in `build_app_and_options_from_config'
	from /var/lib/gems/3.1.0/gems/rack-2.2.9/lib/rack/server.rb:249:in `app'
	from /var/lib/gems/3.1.0/gems/rack-2.2.9/lib/rack/server.rb:422:in `wrapped_app'
	from /var/lib/gems/3.1.0/gems/rack-2.2.9/lib/rack/server.rb:312:in `block in start'
	from /var/lib/gems/3.1.0/gems/rack-2.2.9/lib/rack/server.rb:379:in `handle_profiling'
	from /var/lib/gems/3.1.0/gems/rack-2.2.9/lib/rack/server.rb:311:in `start'
	from /var/lib/gems/3.1.0/gems/rack-2.2.9/lib/rack/server.rb:168:in `start'
	from /var/lib/gems/3.1.0/gems/rack-2.2.9/bin/rackup:5:in `<top (required)>'
	from /usr/local/bin/rackup:25:in `load'
	from /usr/local/bin/rackup:25:in `<top (required)>'
	from /var/lib/gems/3.1.0/gems/bundler-2.5.10/lib/bundler/cli/exec.rb:58:in `load'
	from /var/lib/gems/3.1.0/gems/bundler-2.5.10/lib/bundler/cli/exec.rb:58:in `kernel_load'
	from /var/lib/gems/3.1.0/gems/bundler-2.5.10/lib/bundler/cli/exec.rb:23:in `run'
	from /var/lib/gems/3.1.0/gems/bundler-2.5.10/lib/bundler/cli.rb:455:in `exec'
	from /var/lib/gems/3.1.0/gems/bundler-2.5.10/lib/bundler/vendor/thor/lib/thor/command.rb:28:in `run'
	from /var/lib/gems/3.1.0/gems/bundler-2.5.10/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
	from /var/lib/gems/3.1.0/gems/bundler-2.5.10/lib/bundler/vendor/thor/lib/thor.rb:527:in `dispatch'
	from /var/lib/gems/3.1.0/gems/bundler-2.5.10/lib/bundler/cli.rb:35:in `dispatch'
	from /var/lib/gems/3.1.0/gems/bundler-2.5.10/lib/bundler/vendor/thor/lib/thor/base.rb:584:in `start'
	from /var/lib/gems/3.1.0/gems/bundler-2.5.10/lib/bundler/cli.rb:29:in `start'
	from /var/lib/gems/3.1.0/gems/bundler-2.5.10/exe/bundle:28:in `block in <top (required)>'
	from /var/lib/gems/3.1.0/gems/bundler-2.5.10/lib/bundler/friendly_errors.rb:117:in `with_friendly_errors'
	from /var/lib/gems/3.1.0/gems/bundler-2.5.10/exe/bundle:20:in `<top (required)>'
	from /usr/local/bin/bundle:25:in `load'
	from /usr/local/bin/bundle:25:in `<main>'`
	
Would be great to get this running.
	
Also there seems to be an issue with fluent-package (v5.0.3), at least for Debian12 as the systemd-unit always fails to start the service. There are no meaningful error logs due to fluentd not even starting up to create logs.
